// @iplab/ngx-color-picker/esm2020/lib/components/parts/color-preset/color-preset.component.mjs
import { Component, HostBinding, Input, EventEmitter, Output, ChangeDetectionStrategy } from '@angular/core';
import { Subject, of, fromEvent, merge } from 'rxjs';
import { takeUntil, delay, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./../../../services/color-picker.service";
export class ColorPresetComponent {
    constructor(pickerConfig, elementRef) {
        this.pickerConfig = pickerConfig;
        this.elementRef = elementRef;
        this.selectionChange = new EventEmitter(false);
        this.longPress = new EventEmitter(false);
        this.mouseup = new Subject();
        this.showDepthText = false;
        this.subscriptions = [];
        this.addEventListeners();
    }
    set depth(showDepthText) {
        this.showDepthText = !!showDepthText;
    }
    ngOnDestroy() {
        this.mouseup.next();
        this.mouseup.complete();
        this.removeEventListeners();
    }
    get bgColor() {
        return this.color.toRgbaString();
    }
    get title() {
        const color = this.color ? this.color.toHexString() : '';
        if (this.showDepthText) {
            return this.pickerConfig.presetsTitle.replace(/\{\s*(.+?)\s*\}/g, (match, firstMatch) => color);
        }
        return color;
    }
    get className() {
        return this.activeColor ? this.color.toRgbaString() === this.activeColor.toRgbaString() : false;
    }
    addEventListeners() {
        this.subscriptions.push(merge(fromEvent(this.elementRef.nativeElement, 'mouseup'), fromEvent(this.elementRef.nativeElement, 'touchend'))
            .subscribe(() => this.onTouchEnd()));
        this.subscriptions.push(merge(fromEvent(this.elementRef.nativeElement, 'mousedown'), fromEvent(this.elementRef.nativeElement, 'touchstart', { passive: true }))
            .subscribe((e) => this.onTouch(e)));
    }
    removeEventListeners() {
        this.subscriptions.forEach((subscription) => subscription.unsubscribe());
        this.subscriptions.length = 0;
    }
    onTouch(event) {
        of(event)
            .pipe(map((e) => e.timeStamp || new Date().getTime()), delay(350), takeUntil(this.mouseup))
            .subscribe(() => this.longPress.next(true));
        this.selectionChange.emit(this.color);
    }
    onTouchEnd() {
        this.mouseup.next();
    }
}
ColorPresetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.5", ngImport: i0, type: ColorPresetComponent, deps: [{ token: i1.ColorPickerConfig }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component });
ColorPresetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.5", type: ColorPresetComponent, selector: "color-preset", inputs: { activeColor: "activeColor", color: "color", depth: ["show-depth-title", "depth"] }, outputs: { selectionChange: "selectionChange", longPress: "longPress" }, host: { properties: { "style.backgroundColor": "this.bgColor", "attr.title": "this.title", "class.selected": "this.className" } }, ngImport: i0, template: ``, isInline: true, styles: [":host,:host ::ng-deep *{padding:0;margin:0;box-sizing:border-box}\n", ":host{display:inline-block;height:12px;width:12px;position:relative;cursor:pointer;transition:all .2s}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.5", ngImport: i0, type: ColorPresetComponent, decorators: [{
            type: Component,
            args: [{ selector: `color-preset`, template: ``, changeDetection: ChangeDetectionStrategy.OnPush, styles: [":host,:host ::ng-deep *{padding:0;margin:0;box-sizing:border-box}\n", ":host{display:inline-block;height:12px;width:12px;position:relative;cursor:pointer;transition:all .2s}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.ColorPickerConfig }, { type: i0.ElementRef }]; }, propDecorators: { activeColor: [{
                type: Input
            }], color: [{
                type: Input
            }], depth: [{
                type: Input,
                args: ['show-depth-title']
            }], selectionChange: [{
                type: Output
            }], longPress: [{
                type: Output
            }], bgColor: [{
                type: HostBinding,
                args: ['style.backgroundColor']
            }], title: [{
                type: HostBinding,
                args: ['attr.title']
            }], className: [{
                type: HostBinding,
                args: ['class.selected']
            }] } });
